coderd/notifications/types/payload.go

```diff
@@ -17,4 +17,5 @@ type MessagePayload struct {
 
 	Actions []TemplateAction  `json:"actions"`
 	Labels  map[string]string `json:"labels"`
+	Data    map[string]any    `json:"data,omitempty"`
```

Comment:
For interface stability.

```suggestion
	Data    map[string]any    `json:"data"`
```

Reply:
Can you confirm that it will not affect backward compatibility with existing notifications?

Reply:
You're only adding an extra field, so I believe that will not cause any BC-breaks.